elk/training/train.py

```diff
@@ -181,6 +181,7 @@ def train(cfg: RunConfig, out_dir: Optional[Path]):
                     device=devices[0],
                     out_dir=out_dir,
                 )
+                writer.writerow([L - i] + [f"{s:.4f}" for s in layer_stat])
```

Comment:
i'll like to refactor this reporter functions so that they give back something like a `EvalCSVResult` eventually.

```
from elk.training.reporter import EvalResult


class EvalCSVResult:
    layer: int
    train_loss: float
    eval_result: EvalResult
    # Only available if reporting baseline
    lr_auroc: Optional[float]
    # Only available if reporting baseline
    lr_acc: Optional[float]
```

and then another function does the responsbility of writing to a CSV / rounding to 4 d.p
but i'll do that in another MR :)

Reply:
Sounds good, could you make a new issue for this?